internal/tasks/device_disconnected.go

```diff
@@ -47,17 +57,22 @@ func (t *DeviceDisconnected) Poll(ctx context.Context) {
 
 		var batch []string
 		for _, device := range devices.Items {
+			// Update the device status on the service side
```

Comment:
It seems that the function UpdateServiceSideDeviceStatus updates the device in memory, but the change is not saved to the database.  If not then the same operation will be done again during the next iteration

Reply:
It's saved to the DB below - UpdateDeviceSummaryStatusBatch()

Reply:
OK, this is a pre-existing bug.  Good catch!